src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php

```diff
@@ -177,8 +177,9 @@ public function getConfigTreeBuilder()
                     ->addDefaultsIfNotSet()
                     ->children()
                         ->arrayNode('paths')
-                            ->prototype('scalar')->end()
-                        ->end()
+                        ->prototype('scalar')->end()
+                    ->end()
+                        ->booleanNode('scan_dir')->defaultFalse()->end()
```

Comment:
I don't think we should introduce a new config option. Instead, we should reuse `api_platform.mapping.paths`. Having so many similar options is confusing. We can chain this new extractor with the lowest priority. If necessary, we could provide a toggle to enable/disable this feature.

Reply:
You just want to detect the file? I agree with the option that it's potentially too much. After that if you want I can detect if there is the presence of @ApiResource and automatically disable this option. No ?

Reply:
It's possible to use different extractors together, e.g. using annotations together with YAML / XML. That should not be a problem. We do not need to detect the `@ApiResource` annotation. What's important is to make sure that we **do not override** any metadata provided by the user.